pkg/codegen/pcl/utilities.go

```diff
@@ -56,10 +57,25 @@ func DecomposeToken(tok string, sourceRange hcl.Range) (string, string, string,
 	return components[0], components[1], components[2], nil
 }
 
+func hasDependencyOn(a, b Node) bool {
+	for _, d := range a.getDependencies() {
+		if d.Name() == b.Name() {
+			return true
+		}
+	}
+	return false
+}
+
+func mutuallyDependant(a, b Node) bool {
+	return hasDependencyOn(a, b) && hasDependencyOn(b, a)
+}
+
 func linearizeNode(n Node, done codegen.Set, list *[]Node) {
 	if !done.Has(n) {
 		for _, d := range n.getDependencies() {
-			linearizeNode(d, done, list)
+			if !mutuallyDependant(n, d) {
+				linearizeNode(d, done, list)
+			}
```

Comment:
`linearizeNode` would otherwise call itself infinitely so we need to break it for mutually dependant components